File: src/Bioinformatics Stronghold/Dynamic Programming/004 Recurrence Relations/recursion.py
def main():
    n, k = 32, 3
    Solution = rabbit_recur(n, k)
    print(f"{Solution = }")

def rabbit_recur(months: int, offspring: int) -> int:
    if months <= 0:
        return 0
    if months == 1:
        return 1
    return rabbit_recur(months-1, offspring)+(rabbit_recur(months-2, offspring)*offspring)

if __name__ == "__main__":
    main()

File: src/Bioinformatics Stronghold/Dynamic Programming/004 Recurrence Relations/recursion.c
#include <stdio.h>

int rabbit_recur(int months, int offspring);

int main(){
    int n = 32;
    int k = 3;
    int rabbit = rabbit_recur(n, k);
    printf("%d\n", rabbit);
    return 0;
}

int rabbit_recur(int months, int offspring){
    if(months <= 0) return 0;
    if(months == 1) return 1;
    return rabbit_recur(months-1, offspring)+(rabbit_recur(months-2, offspring)*offspring);
}
